### PR TITLE
[DWG] :  add DWG_ALL_ATTRIBUTES to get all attributes #5103

### PR DESCRIPTION
**Pull request by lchevallier69**
display all attribute without taking visibility into account
## What does this PR do?
add a new config parameter DWG_ALL_ATTRIBUTES to ignore or expose unvisible attributes
default value is set to TRUE, not only visible attributes should be exposed, visibility is only for display
## What are related issues/pull requests?
#5103

### CODE REVIEW

File: autotest/ogr/ogr_dwg.py [hunk 1]
```diff
@@ -177,4 +177,70 @@ def test_ogr_dwg_4():
...
+        layer = ds.GetLayer( 'entities' )
+        defn = layer.GetLayerDefn()
+
+        assert defn.GetFieldCount() == 30, \
+            ('did not get expected number of fields in defn. got %d'
+                                % defn.GetFieldCount())
+
+        ds = None
+
+def test_ogr_dwg_5():
```
Comment (rouault) [inline autotest/ogr/ogr_dwg.py:213]:
```suggestion
def test_ogr_dwg_6():
```

**Comment by rouault** [comment]
due to the duplicated test_ogr_dwg_5(), that I've committed, the first test_ogr_dwg_5 fails with:
```
>           assert defn.GetFieldCount() == 30, \
                ('did not get expected number of fields in defn. got %d'
                                    % defn.GetFieldCount())
E           AssertionError: did not get expected number of fields in defn. got 32
E           assert 32 == 30
E            +  where 32 = <bound method FeatureDefn.GetFieldCount of <osgeo.ogr.FeatureDefn; proxy of <Swig Object of type 'OGRFeatureDefnShadow *' at 0x7f045c1ea9f0> >>()
E            +    where <bound method FeatureDefn.GetFieldCount of <osgeo.ogr.FeatureDefn; proxy of <Swig Object of type 'OGRFeatureDefnShadow *' at 0x7f045c1ea9f0> >> = <osgeo.ogr.FeatureDefn; proxy of <Swig Object of type 'OGRFeatureDefnShadow *' at 0x7f045c1ea9f0> >.GetFieldCount
```

But as far as I can test_ogr_dwg_5 and test_ogr_dwg_6 do the same thing, but in a reversed order. Some cleanup needed...

**Comment by lchevallier69** [comment]
ok my mistake, test_ogr_dwg_5 may do 2 tests => one with the new parameter FALSE and one with no parameter to chack default value wich is TRUE. This tests contains two different config value. Duplicate code is in fact a merge branch problem form my internal feature branch. I Recheck
Thanks

**Comment by rouault** [comment]
@lchevallier69 any progress on finalizing this PR ?

**Comment by lchevallier69** [comment]
Yep, tests are clean nox

**Comment by rouault** [comment]
/__w/gdal/gdal/doc/source/drivers/vector/dwg.rst:73: WARNING: Title underline too short.
/__w/gdal/gdal/doc/source/drivers/vector/dwg.rst:77: ERROR: Unexpected indentation.

**Pull request merged**